pkg/flare/archive.go

```diff
@@ -621,6 +626,45 @@ func zipSecrets(tempDir, hostname string) error {
 	return writeScrubbedFile(f, b.Bytes())
 }
 
+func zipProcessChecks(tempDir, hostname string, getAddressPort func() (url string, err error)) error {
+	addressPort, err := getAddressPort()
+	if err != nil {
+		return fmt.Errorf("wrong configuration to connect to process-agent")
+	}
+	checkURL := fmt.Sprintf("http://%s", addressPort)
+
+	err = zipHTTPCallContent(tempDir, hostname, "process_check_output.json", checkURL+"/check/process")
+	if err != nil {
+		_ = log.Error(err)
+		err = ioutil.WriteFile(
+			filepath.Join(tempDir, hostname, "process_check_output.json"),
+			[]byte("error: process-agent is not running or is unreachable"),
+			os.ModePerm,
+		)
+		if err != nil {
+			return err
+		}
+	}
+	if err != nil {
+		return fmt.Errorf("could not connect to process-agent: %v", err)
+	}
+
+	err = zipHTTPCallContent(tempDir, hostname, "container_check_output.json", checkURL+"/check/container")
+	if err != nil {
+		_ = log.Error(err)
+		err = ioutil.WriteFile(
+			filepath.Join(tempDir, hostname, "container_check_output.json"),
+			[]byte("error: process-agent is not running or is unreachable"),
+			os.ModePerm,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
```

Comment:
We could probably consolidate these with a helper - the different part is the name of the check and the path in the flare? Could you also please add `process_discovery`. Could we do these based on the enabled settings in the config? E.g. only collect process check output when it is enabled.

Reply:
There's no reason why we couldn't do both of those things.

The current behavior is to just return a 404 with an error message in the body if the check has not been run, but I think it would be fine to just omit those files